odl/space/pspace.py

```diff
@@ -629,6 +664,22 @@ def __setitem__(self, indices, values):
             for i, index in enumerate(indices):
                 self.parts[index] = values[i]
 
+    __add__ = _broadcast_arithmetic('__add__')
+    __radd__ = _broadcast_arithmetic('__radd__')
+    __iadd__ = _broadcast_arithmetic('__iadd__')
+    __sub__ = _broadcast_arithmetic('__sub__')
+    __rsub__ = _broadcast_arithmetic('__rsub__')
+    __isub__ = _broadcast_arithmetic('__isub__')
+    __mul__ = _broadcast_arithmetic('__mul__')
+    __rmul__ = _broadcast_arithmetic('__rmul__')
+    __imul__ = _broadcast_arithmetic('__imul__')
+    __div__ = _broadcast_arithmetic('__div__')
+    __rdiv__ = _broadcast_arithmetic('__rdiv__')
+    __idiv__ = _broadcast_arithmetic('__idiv__')
+    __truediv__ = _broadcast_arithmetic('__truediv__')
+    __rtruediv__ = _broadcast_arithmetic('__rtruediv__')
+    __itruediv__ = _broadcast_arithmetic('__itruediv__')
```

Comment:
could hypthetically be written as a loop:

```python
for op in ['add', 'sub', 'mul', 'div', 'truediv']:
    for modifier in ['', 'r', 'i']:
        name = '__{}{}__'.format(modifier, op)
        setattr(ProductSpaceVector(name), _broadcast_arithmetic(name))
```

I'm not sure if this is better or not.